packages/template-retail-react-app/app/hooks/use-address-form.js

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2025, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {useState, useCallback, useMemo} from 'react'
+import {useForm} from 'react-hook-form'
+import {useToast} from '@salesforce/retail-react-app/app/hooks/use-toast'
+import {useIntl} from 'react-intl'
+import {useShopperCustomersMutation} from '@salesforce/commerce-sdk-react'
+import {useCurrentCustomer} from '@salesforce/retail-react-app/app/hooks/use-current-customer'
+import {
+    areAddressesEqual,
+    sanitizedCustomerAddress
+} from '@salesforce/retail-react-app/app/utils/address-utils'
+import {nanoid} from 'nanoid'
+
+export const useAddressForm = (
+    addGuestAddress,
+    isGuest,
+    setAddressesForItems,
+    availableAddresses,
+    deliveryItems
+) => {
+    const {formatMessage} = useIntl()
+    const showToast = useToast()
+    const {data: customer, refetch: refetchCustomer} = useCurrentCustomer()
+    const [showForm, setShowForm] = useState({})
```

Comment:
the name `showForm`, i thought it is a bool. What do you think changing it to formMap or formByItemId? 

```suggestion
    const [showForm, setShowForm] = useState({})
```